### PR TITLE
Feature/enter building

### PR DESCRIPTION
**Pull request by tomermoshe**

### CODE REVIEW

File: packages/client/src/app/game/services/game-settings.service.ts [hunk 1]
```diff
@@ -3,4 +3,16 @@ import { Injectable } from '@angular/core';
 @Injectable()
```
Comment (eitanfr) [inline packages/client/src/app/game/services/game-settings.service.ts]: remove

File: packages/client/src/app/game/services/collision-detector.service.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+      this._collision = this.character.enternedBuilding ?
+        this.checkCollisionInRoom(fromLocation) :
+        this.getDepthDistance(fromLocation, this.windowCenter) < CollisionDetectorService.COLLIDE_FACTOR_METER;
+    }
+
+    if (this._collision) {
+      this.characterHeadingWhenCollided = this.character.heading;
+      if (!this.character.enternedBuilding) {
+        const pickedFeature = this.viewer.scene.pick(this.windowCenter, 300, 300);
+        if (pickedFeature && !pickedFeature.mesh && pickedFeature._batchId) {
```
Comment (eitanfr) [inline packages/client/src/app/game/services/collision-detector.service.ts]: remove mash or use instance of

File: packages/client/src/app/game/views/game-map/keyboard-control/keyboard-control.component.ts [hunk 3]
```diff
@@ -177,6 +154,27 @@ export class KeyboardControlComponent implements OnInit {
           this.changeViewMove();
         });
       });
+    this.keyboardKeysService.registerKeyBoardEvent('KeyE', 'Enter Nearby Building',
+      (keyEvent: KeyboardEvent) => {
+        if (this.character.enternedBuilding) {
+          this.character.tileBuilding.show = true;
+          this.character.location = this.character.enteringBuildingPosition;
+          this.buildingsService.removeBuilding(this.character.enternedBuilding);
```
Comment (eitanfr) [inline packages/client/src/app/game/views/game-map/keyboard-control/keyboard-control.component.ts]: extract to func

File: packages/client/src/app/game/views/game-map/keyboard-control/keyboard-control.component.ts [hunk 4]
```diff
@@ -177,6 +154,27 @@ export class KeyboardControlComponent implements OnInit {
...
+          this.character.enternedBuilding = undefined;
+          this.character.tileBuilding = undefined;
+          this.character.enteringBuildingPosition = undefined;
+          this.character.nearbyBuildingPosition = undefined;
+          this.gameService.updateServerOnPosition(true);
+
+        } else if (this.character.nearbyBuildingPosition) {
+          this.character.tileBuilding.show = false;
+          this.character.enternedBuilding = this.buildingsService.createBuilding(this.character.nearbyBuildingPosition);
```
Comment (eitanfr) [inline packages/client/src/app/game/views/game-map/keyboard-control/keyboard-control.component.ts]: extrcat to func

File: packages/client/src/app/game/views/game-map/me/me.component.ts [hunk 5]
```diff
@@ -81,8 +93,8 @@ export class MeComponent implements OnInit, OnDestroy {
   }
 
   private showGunMuzzleFlash() {
-    this.isMuzzleFlashShown = true;
-    setTimeout(() => this.isMuzzleFlashShown = false, 20);
+    this.muzzleFlash.nativeElement.style.visibility = 'visible';
```
Comment (eitanfr) [inline packages/client/src/app/game/views/game-map/me/me.component.ts]: why? 
you can use [hidden]

File: packages/client/src/app/game/services/collision-detector.service.ts (hunk 2 above)
Comment (tomermoshe) [reply packages/client/src/app/game/services/collision-detector.service.ts]: done

File: packages/client/src/app/game/services/game-settings.service.ts (hunk 1 above)
Comment (tomermoshe) [reply packages/client/src/app/game/services/game-settings.service.ts]: done

File: packages/client/src/app/game/views/game-map/me/me.component.ts (hunk 5 above)
Comment (tomermoshe) [reply packages/client/src/app/game/views/game-map/me/me.component.ts]: done

File: packages/client/src/app/game/views/game-map/keyboard-control/keyboard-control.component.ts (hunk 3 above)
Comment (tomermoshe) [reply packages/client/src/app/game/views/game-map/keyboard-control/keyboard-control.component.ts]: done

File: packages/client/src/app/game/views/game-map/keyboard-control/keyboard-control.component.ts (hunk 4 above)
Comment (tomermoshe) [reply packages/client/src/app/game/views/game-map/keyboard-control/keyboard-control.component.ts]: done

**Pull request merged**